header-link.js

```diff
@@ -0,0 +1,6 @@
+let elements = document.getElementsByClassName("header-copy");
+for (let i = 0; i < elements.length; i++) {
+    let element = elements[i];
+    let parentId = element.parentElement.id;
+    addCopyButton(element, location.protocol + '//' + location.host + location.pathname + "#" + parentId)
+}

```

Comment:
```suggestion
const elements = document.getElementsByClassName("header-copy");
for (let element of elements) {
    let parentId = element.parentElement.id;
    element.onclick = function() {
        navigator.clipboard.writeText(location.protocol + '//' + location.host + location.pathname + "#" + parentId);
        history.pushState({}, document.title, "#" + parentId);
    }
}
```

This might be a bit prettier, I think the alert which has to be dismissed is a bit hindering, (a self-dismissing toast/popup message might be a nice solution, but is also much more work) and replacing the hash in the current url is probably expected by most users (with `history.pushState` so that the browser doesn't scroll).

What do you think?

Reply:
1. There doesn't seem to be any consensus on if it should copy, scroll or both across the internet. I think just replacing the hash would be a bit confusing though. We can talk about it more in discord
2. We should use the `addCopyButton` function and just add a toast thingy to that, i think that would be best for everything. There needs to be some feedback so people know whats happening